proto/opamp.proto

```diff
@@ -802,6 +802,11 @@ message PackageStatus {
 
     // Error message if the status is erroneous.
     string error_message = 7;
+
+    // Optional details that may be of interest to a user.
+    // For example the download rate, percent downloaded, or a message.
+    // Status: [Development]
+    map<string, string> metadata = 8;
```

Comment:
I am not sure about this one. Since we don't define what the metadata is going to be it becomes purely agent-specific. When possible we should avoid agent-specific additions since it means servers have to implement agent-specific logic. The dowloading of a package seems to be a common concern, so I don't see at the moment why it needs to be an open map with "anything possible to record" approach.

I would like to see use cases that demonstrate that every agent is going to use different metadata before we can go ahead with this and there is nothing or very little common between agent types.

Absent the use cases I suggest we do one of the following:
1. Delete this from the PR so that we can merge it with the other change, which is not controversial.
1. Replace this by a more concrete set of details that describe the state of downloading (e.g. bytes downloaded, ETA, etc).